src/Compilers/CSharp/Portable/Symbols/Source/SourceOrdinaryMethodSymbol.cs

```diff
@@ -721,7 +721,7 @@ private static void PartialMethodConstraintsChecks(SourceOrdinaryMethodSymbol de
             }
 
             var typeParameters2 = implementation.TypeParameters;
-            var indexedTypeParameters = IndexedTypeParameterSymbol.Take(arity);
+            var indexedTypeParameters = TypeMap.TypeParametersAsTypeSymbolsWithIgnoredAnnotations(IndexedTypeParameterSymbol.Take(arity));
```

Comment:
>Take [](start = 125, length = 4)

Consider changing what this method returns instead. It looks like majority of call sites actually need TypeWithAnnotations. For the case of just type parameters, we can add a special helper.

Reply:
I was intentionally suggesting to keep the same name. There are often overloads that can handle arrays of both types, however we want default case to be TypeWithAnnotations with an ignored annotation. This way, every consumer of this method, existing or added later will be automatically guided into the right direction and will not require changes. Changes will be required only for exceptions, which aren't that many.  

---
In reply to: [535737016](https://github.com/dotnet/roslyn/pull/49723#discussion_r535737016) [](ancestors = 535737016)